tool-support/syntax-highlighting/vscode/sysml/syntaxes/sysml.tmLanguage.json

```diff
@@ -18,11 +18,11 @@
   "repository": {
     "keywords": {
       "patterns": [
-        { "match": "\\b(about|abstract|accept|action|actor|after|alias|all|allocate|allocation|analysis|and|as|assert|assign|assume|at|attribute|bind|binding|by|calc|case|comment|concern|connect|connection|constraint|decide|def|default|defined|dependency|derived|do|doc|else|end|entry|enum|event|exhibit|exit|expose|false|feature|filter|first|flow|for|fork|frame|from|hastype|if|implies|import|in|include|individual|inout|interface|istype|item|join|language|loop|merge|message|metadata|nonunique|not|null|objective|occurrence|of|or|ordered|out|package|parallel|part|perform|port|private|protected|public|readonly|redefines|ref|render|rendering|rep|require|requirement|return|satisfy|send|snapshot|specializes|stakeholder|state|subject|subsets|succession|then|timeslice|to|transition|true|until|use|variant|variation|verification|verify|via|view|viewpoint|when|while|xor)\\b", "name": "keyword.other.sysml" },
+        { "match": "\\b(about|abstract|accept|action|actor|after|alias|all|allocate|allocation|analysis|and|as|assert|assign|assume|at|attribute|bind|binding|by|calc|case|comment|concern|connect|connection|constraint|decide|def|default|defined|dependency|derived|do|doc|else|end|entry|enum|event|exhibit|exit|expose|false|filter|first|flow|for|fork|frame|from|hastype|if|implies|import|in|include|individual|inout|interface|istype|item|join|language|loop|merge|message|metadata|nonunique|not|null|objective|occurrence|of|or|ordered|out|package|parallel|part|perform|port|private|protected|public|readonly|redefines|ref|render|rendering|rep|require|requirement|return|satisfy|send|snapshot|specializes|stakeholder|state|subject|subsets|succession|then|timeslice|to|transition|true|until|use|variant|variation|verification|verify|via|view|viewpoint|when|while|xor)\\b", "name": "keyword.other.sysml" },
         { "match": "(%|&|&&|\\?|\\?\\?|\\^|\\^\\^|\\||\\|\\|)", "name": "keyword.operator.logical.sysml" },
         { "match": "(!|!=|\\<|\\<=|=|==|\\>|\\>=)", "name": "keyword.operator.comparison.sysml" },
         { "match": "(\\*|\\*\\*|\\+|\\-|/)", "name": "keyword.operator.arithmetic.sysml" },
-        { "match": "(\\-\\>|\\.\\.|:|::|:=|:\\>|:\\>\\>|=\\>|@|~)", "name": "keyword.operator.other.sysml" }
+        { "match": "(\\-\\>|\\.$|\\.\\.|\\.\\?|:|::|:=|:\\>|:\\>\\>|=\\>|@|~)", "name": "keyword.operator.other.sysml" }
```

Comment:
The `$` char should be escaped as `\\$`. Small adaptation needed in `tool-support/syntax-highlighting/xtext_grammar_converter.py` line 409, to properly recognize the `.$` operator, as `$` is end of test string symbol in regex.

Reply:
We actually decided use `.?` rather than `.$` in the expression syntax. The appearance of `.$` here is do to this not being properly updated in the SysML.xtext grammar for `TargetExpression`. Good that we found this! I will fix it on this branch.